### ISSUE TITLE
Calculate the number of spares within a catalogue item

### ISSUE THREAD

**Issue by joelvdavies** [ral-facilities/inventory-management-system-api#417]
Following on from #413 and #416. Update the `spares` field in the parent catalogue item whenever:

- [x] An item is created
- [x] An item is deleted
- [x] ~An item's usage status is updated~ An item's system_id is updated
- [ ] ~The definition of a spare is updated~ Not needed now as cannot set until after #547 

It is not currently possible to move items between catalogue items. The calculation itself should:

- [x] Be done in a transaction - editing the spares field first to ensure write locked to prevent multiple conflicting updates (also need the inserts/update/delete to be done in the same transaction as well to ensure they cannot become out of sync)
- [x] Consider https://stackoverflow.com/questions/60237515/mongodb-returning-wrong-count
- [x] Raise an appropriate error like #412 if a conflict occurs (https://stackoverflow.com/questions/65886341/mongo-throwing-write-conflicts-for-concurrent-sessions - For now we wont retry the database as should be extremely rare and avoids a `time.sleep`)
